nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -0,0 +1,493 @@
+
+        return processed_example
+
+
+def convert_canary_prompt_to_text(prompt, is_canary_tokens_augment):
+    ps = prompt.replace("<pad>", "").split('>')
+
+    def get_lang(text):
+        if text == "<|fr|":
+            lang = 'French'
+        elif text == "<|es|":
+            lang = 'Spanish'
+        elif text == "<|de|":
+            lang = 'German'
+        elif text == "<|en|":
+            lang = 'English'
+        else:
+            assert False, 'Unknown language {}'.format(prompt)
+        return lang
+
+    def get_task_template(text, is_canary_tokens_augment, simple_augment=True):
+        if text == "<|transcribe|":
+            template = 'Transcribe the spoken content to written <|SLANG|> text, <|PNC|>.'
+        elif text == "<|translate|":
+            if is_canary_tokens_augment:
+                if simple_augment:
+                    template = 'Transcribe the spoken content to written <|SLANG|> text, then translate this to <|TLANG|> text, <|PNC|>'
+                else:
+                    template = 'Transcribe the spoken content to written <|SLANG|> text, then translate this to English text, then translate this to <|TLANG|> text, <|PNC|>'
+            else:
+                template = 'Translate the spoken <|SLANG|> content to written <|TLANG|> text, <|PNC|>'
+        else:
+            assert False, 'Unknown task {}'.format(prompt)
+        return template
+
+    def get_pnc(text):
+        if text == "<|nopnc|":
+            pnc = 'ignoring punctuations and capitalization'
+        elif text == "<|pnc|":
+            pnc = 'with punctuations and capitalizations'
+        else:
+            assert False, 'Unknown pnc {}'.format(prompt)
+        return pnc
+
+    if len(ps) == 6:
+        source_lang = get_lang(ps[1])
+        target_lang = get_lang(ps[3])
+        pnc = get_pnc(ps[4])
+        task = get_task_template(ps[2], is_canary_tokens_augment)
+        task = task.replace('<|SLANG|>', source_lang)
+        task = task.replace('<|TLANG|>', target_lang)
+        task = task.replace('<|PNC|>', pnc)
+    else:
+        task = 'Above is not speech.'
+    return task
+
+
+class LhotseAudioQuestionAnswerDataset(torch.utils.data.Dataset):
+    """
+    This dataset is based on Lhotse ASR dataset from ``audio_to_text_lhotse.py``
+    and ``TarredAudioQuestionAnswerDataset`` from ``audio_text_qa_dataset.py``.
+
+    Unlike native NeMo datasets, Lhotse dataset defines only the mapping from
+    a CutSet (meta-data) to a mini-batch with PyTorch tensors.
+    Specifically, it performs tokenization, I/O, augmentation, and feature extraction (if any).
+    Managing data, sampling, de-duplication across workers/nodes etc. is all handled
+    by Lhotse samplers instead.
+    """
+
+    def __init__(
+        self,
+        text_processor: TextProcessing,
+        default_question: str,
+        tokens_to_generate: int,
+        pad_to_max_length: bool,
+        max_seq_length: int,
+        noise_cuts: Optional = None,
```

Comment:
this is a left-over from before lhotse nemo dataset was refactored; remove noise_cuts in this class.

Reply:
Done